### ISSUE TITLE
[ARCHETYPE-565] Unable to resolve groovy.json classes when planting

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-archetype#572]
**[Edgar Cantu](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ecantu)** opened **[ARCHETYPE-565](https://issues.apache.org/jira/browse/ARCHETYPE-565?redirect=false)** and commented

With the latest release of the archetype generator, projects that have a post-generation script that import from other packages not in groovy core fail to build with an error similar to the following:

```bash
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-archetype-plugin:3.1.0:generate (default-cli) on project standalone-pom: startup failed:
[ERROR] Script1.groovy: 1: unable to resolve class groovy.json.JsonException
[ERROR]  @ line 1, column 1.
[ERROR]    import groovy.json.JsonException
[ERROR]    ^
[ERROR]
```

This used to work in the previous version: 3.0.1 of the archetype-plugin.

Suspected change that introduced this backwards incompatibility:
https://github.com/apache/maven-archetype/commit/11cb603da9d3da813c0a34a9e2532cc2c85f940b#diff-a6aa95e2d0a1da7b4460e2506c2e67daR56

Essentially, with https://issues.apache.org/jira/browse/ARCHETYPE-536 it looks like the `groovy` dependency went from 1.8.3 to 2.4.16 and groovy split modules from groovy core with the 2.x release. There is a `groovy-all` package that could be used.

---

**Affects:** 3.1.0

**Issue Links:**
- [ARCHETYPE-683](https://issues.apache.org/jira/browse/ARCHETYPE-683) [REGRESSION] groovy.json.JsonOutput not found

**Remote Links:**
- [GitHub Pull Request #58
](https://github.com/apache/maven-archetype/pull/58)

2 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Josh Reagan](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jreagan)** commented

https://github.com/apache/maven-archetype/pull/58

**Comment by jira-importer** [comment]
**[Sylwester Lachiewicz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slachiewicz)** commented

Merged with [b3ef1466d0d075f97d7690d4980c976233338f4c](https://gitbox.apache.org/repos/asf?p=maven-archetype.git;a=commit;h=b3ef1466d0d075f97d7690d4980c976233338f4c)
